docs/backups-api.yaml

```diff
@@ -0,0 +1,213 @@
+openapi: 3.0.3
+info:
+  title: Dataset Backup API
+  description: API for managing dataset backups, including creation, listing, restoration, and deletion.
+  version: 0.82.8
+servers:
+  - url: http://localhost:3030/
+    description: Base path for dataset backup operations.
+
+paths:
+  /backups/create/{datasetName}:
+    post:
+      summary: Create a backup for a dataset
+      description: Initiates a backup operation for the specified dataset.
+      parameters:
+        - name: datasetName
+          in: path
+          required: true
+          description: The name of the dataset to back up.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Backup created successfully.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  dataset:
+                    type: string
+                    description: Name of the dataset backed up.
+                  date:
+                    type: string
+                    format: date-time
+                    description: Timestamp of the backup operation.
+                  backup:
+                    type: object
+                    description: Details of the backup operation.
+        '500':
+          description: Internal server error.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+                    description: Error message.
+  /backups/create/:
+    post:
+      summary: Create a backup for all available datasets
+      description: Initiates a backup operation for all of datasets .
+      responses:
+        '200':
+          description: Backup created successfully.
+          content:
+            application/json:
+              schema:
+                type: object
```

Comment:
Should this not be a list of objects in this case as there might be multiple datasets backed up?

Reply:
It's technically a JSON map that contains a list i.e. 

```json
{
  "dataset" : "/",
  "date" : "2024-12-04_16-05-50",
  "backup" : {
    "backup-id" : 2,
    "date" : "2024-12-04_16-05-50",
    "datasets" : [ {
      "dataset-id" : "/ds1",
      "tdb" : {
        "folder" : "/Users/paulgallagher/tmp/crap/data/ABAC-DB6/backup//2/ds1/TDB/",
        "files" : {
          "files" : [ "ds1_backup.nq.gz" ]
        },
        "success" : true
      },
      "labels" : {
        "reason" : "No Label Store to back up (not ABAC)",
        "success" : false
      }
    }, {
      "dataset-id" : "/ds",
      "tdb" : {
        "folder" : "/Users/paulgallagher/tmp/crap/data/ABAC-DB6/backup//2/ds/TDB/",
        "files" : {
          "files" : [ "ds_backup.nq.gz" ]
        },
        "success" : true
      },
      "labels" : {
        "folder" : "/Users/paulgallagher/tmp/crap/data/ABAC-DB6/backup//2/ds/labels/",
        "files" : {
          "meta" : {
            "files" : [ "1" ]
          },
          "shared_checksum" : {
            "files" : [ "000008_s6JD5HSEIWCI9HCVF00UW_13904.sst" ]
          },
          "private" : {
            "1" : {
              "files" : [ "OPTIONS-000034", "CURRENT", "MANIFEST-000032", "000031.log" ]
            }
          }
        },
        "success" : true
      }
    } ]
  }
  ```